File: DataStructure_TA_PondS/Data Structure Algorithm/Source Code/Chapter 5/Java/Chapter5/src/chapter5/InfixtoPostfix.java
package chapter5;

public class InfixtoPostfix {

    public static void main(String[] args) {
        String infix = "a-(b+c*d)/e";
        String postfix = "";
        StackLinkedListChar stack = new StackLinkedListChar();
        char[] ch = infix.toCharArray();
        for (int i = 0; i < ch.length; i++) {
            switch (ch[i]) {
                case '+':
                case '-':
                case '*':
                case '/':
                case '%':
                case '(':
                    stack.push(ch[i]);
                    break;
                case ')':
                    char stackBuffer = stack.pop();
                    while (stackBuffer != '(') {
                        postfix += stackBuffer;
                        stackBuffer = stack.pop();
                    }
                    break;
                default:
                    postfix += ch[i];
            }
        }
        while (!stack.isEmpty()) {
            postfix += stack.pop();
        }
        System.out.println(postfix);
    }
}

File: DataStructure_TA_PondS/Data Structure Algorithm/Source Code/Chapter 5/C/InfixtoPostfix.cpp
#include <stdio.h>
#include <conio.h>
#include <StackLinkedListChar.h>

void main(){
    char infix[] = "a-(b+c*d)/e";
    int length = sizeof(infix);
    char postfix[50];
    int countpostfix = 0;

    for (int i=0; i<length-1; i++){
      if((infix[i]=='+')||(infix[i]=='-')|| (infix[i]=='*')||(infix[i]=='/')||(infix[i]=='%')||(infix[i]=='(')){
        push(infix[i]);
      }else if(infix[i] == ')'){
        char stackBuffer = pop();
        while (stackBuffer != '('){
          postfix[countpostfix] = stackBuffer;
          countpostfix++;
          stackBuffer = pop();
        }
      }else{
        postfix[countpostfix] = infix[i];
        countpostfix++;
      }
    }
    while (isEmpty() == false){
      char buff = peek();
      postfix[countpostfix] = pop();
      countpostfix++;
    }
    postfix[countpostfix] = '\0';
    printf("Infix:%s\n",infix);
    printf("Postfix:%s",postfix);

    int n;
    scanf("%d",&n);
}

